### PR TITLE
feat(tests): adds integration tests for views

### LINKED ISSUE #9714: Create tests for view existence
To avoid things like #9690, we could write tests to verify view existence for all views present at 2.0 release. Shouldn't be a problem to dump the register and use it as dataProvider. The only question is plugin views. We probably need to enable all bundled plugins for tests (should be worth it to catch other issues as well)

### PR DESCRIPTION
**Pull request by hypeJunction**
Views are now tested for existence and syntax errors.

Fixes #9714
Refs #9212

### CODE REVIEW

**Comment by hypeJunction** [comment]
Honestly, test results are complete WTF.

**Comment by hypeJunction** [comment]
For integration testing to work properly plugins will need to return a closure, otherwise we loose plugin init calls. I guess we can just add it to testing docs - plugins only have to do it if they play to write integration tests.

File: engine/classes/Elgg/Application.php [hunk 1]
```diff
@@ -261,6 +263,14 @@ public function bootCore() {
 		// in case not loaded already
 		$this->loadCore();
 
+		$hooks = $this->_services->hooks;
```
Comment (hypeJunction) [inline engine/classes/Elgg/Application.php]: I accidentally changed the boot sequence :) Needs to be fixed whether or not this PR makes it in.

Comment (mrclay) [inline engine/classes/Elgg/Application.php]: Can you make an issue for this so it doesn't get lost?

Comment (hypeJunction) [inline engine/classes/Elgg/Application.php]: It's in #11188

**Comment by hypeJunction** [comment]
Replaced by #11195

**Pull request closed without merging**